src/components/Checkbox/Checkbox.tsx

```diff
@@ -188,7 +183,3 @@ export const Checkbox = forwardRef<CheckboxHandles, CheckboxProps>(
 );
 
 function noop() {}
-
-function stopPropagation<E>(event: React.MouseEvent<E>) {
```

Comment:
IIRC this was also part of the focus issu

Reply:
Pretty sure this had to do with the bulk actions. I would remove this.

Reply:
Can you elaborate on the bulk action issue that led us to add this ?

Reply:
Please refer to other PR comments on the `IndexTable` component in which I have added a fix to properly support the bulk actions.

TLDR; the root cause of the issue were the same as for the checkbox hence managing events on other DOM nodes then the originating element.